Interpolation/include/CGAL/interpolation_functions.h

```diff
@@ -73,13 +72,14 @@ linear_interpolation(ForwardIterator first, ForwardIterator beyond,
   typedef typename boost::result_of<ValueFunctor(arg_type)>::type                 result_type;
   typedef typename result_type::first_type                                        Value_type;
 
-  Value_type result(0);
-  result_type val;
+  result_type val = value_function(first->first);
```

Comment:
Need to check that the range is not empty somewhere.

Reply:
I added  preconditions for all interpolation functions in the code and the manual. 
@lrineau: Note that this is not a change as we already now have the precondition that the sum of weights `norm` is larger than zero, which implies a non-empty range.